addons/addon-base-raas/packages/base-raas-services/lib/data-source/access-strategy/roles-only/__tests__/application-role-methods.test.js

```diff
@@ -0,0 +1,338 @@
+  bucket,
+  bucketRegion,
+  status,
+  qualifier,
+  boundaryPolicyArn,
+  studies,
+});
+
+describe('ApplicationRoleMethods', () => {
+  beforeEach(async () => {});
+
+  it('ensures toAppRoleEntity does not fail with undefined dbEntity', async () => {
+    // EXECUTE & CHECK
+    expect(toAppRoleEntity()).toStrictEqual(undefined);
+  });
+
+  it('ensures toAppRoleEntity does not fail with valid dbEntity', async () => {
+    const expectedAppRoleEntity = createAppRole();
+    const dbEntity = {
+      awsPartition: 'aws',
+      boundaryPolicyArn: 'arn:aws:iam::123456789012:policy/swb-IhsKhN8GsLneiis11ujlb8-app-1234567890xxx',
+      bucket: 'bucket-1',
+      pk: 'ACT#1122334455',
+      sk: 'APP#bucket-1#arn:aws:iam::123456789012:role/swb-IhsKhN8GsLneiis11ujlb8-app-1234567890xxx',
+      bucketRegion: 'us-east-1',
+      by: 'sampleUser',
+      mainRegion: 'us-east-1',
+      name: 'swb-IhsKhN8GsLneiis11ujlb8-app-1234567890xxx',
+      qualifier: 'swb-IhsKhN8GsLneiis11ujlb8',
+      status: 'pending',
+      studies: { 'study-1': { accessType: 'readonly', folder: '/', kmsScope: 'none' } },
+    };
+
+    // EXECUTE & CHECK
+    expect(toAppRoleEntity(dbEntity)).toStrictEqual(expectedAppRoleEntity);
+  });
+
+  it('ensures toDbEntity does not fail with valid appRoleEntity', async () => {
+    // BUILD
+    const appRole = createAppRole();
+    const by = 'sampleUserUid';
+    const expectedDbEntity = {
+      arn: 'arn:aws:iam::123456789012:role/swb-IhsKhN8GsLneiis11ujlb8-app-1234567890xxx',
+      awsPartition: 'aws',
+      boundaryPolicyArn: 'arn:aws:iam::123456789012:policy/swb-IhsKhN8GsLneiis11ujlb8-app-1234567890xxx',
+      bucket: 'bucket-1',
+      bucketRegion: 'us-east-1',
+      by: 'sampleUser',
+      createdBy: 'sampleUserUid',
+      mainRegion: 'us-east-1',
+      name: 'swb-IhsKhN8GsLneiis11ujlb8-app-1234567890xxx',
+      qualifier: 'swb-IhsKhN8GsLneiis11ujlb8',
+      status: 'pending',
+      studies: { 'study-1': { accessType: 'readonly', folder: '/', kmsScope: 'none' } },
+      updatedBy: 'sampleUserUid',
+    };
+
+    // EXECUTE & CHECK
+    expect(toDbEntity(appRole, by)).toStrictEqual(expectedDbEntity);
+  });
+
+  it('ensures newAppRoleEntity creates a new valid appRoleEntity', async () => {
+    // BUILD
+    const studyEntity = createStudy();
+    const accountEntity = { mainRegion: 'us-east-1' };
+    const bucketEntity = { kmsArn: 'sampleKmsArn' };
+    const expectedAppRole = {
+      accountId: '1122334455',
+      awsPartition: 'aws',
+      bucket: 'bucket-1',
+      bucketKmsArn: 'sampleKmsArn',
+      bucketRegion: undefined,
+      mainRegion: 'us-east-1',
+      qualifier: 'swb-IhsKhN8GsLneiis11ujlb8',
+      status: 'pending',
+      studies: { 'study-1': { accessType: 'readwrite', folder: '/', kmsArn: undefined, kmsScope: 'none' } },
+    };
+
+    // EXECUTE & CHECK
+    const returnVal = newAppRoleEntity(accountEntity, bucketEntity, studyEntity);
+    expect(returnVal).toHaveProperty('accountId');
```

Comment:
Check on properties that will be validated through `expectedAppRole` should not be necessary.